trinity/chains/__init__.py

```diff
@@ -0,0 +1,108 @@
+import os
+
+from evm.chains.ropsten import ROPSTEN_GENESIS_HEADER
+from evm.p2p import ecies
+from evm.exceptions import CanonicalHeadNotFound
+
+from trinity.constants import (
+    ROPSTEN,
+)
+from trinity.utils.db import (
+    get_chain_db,
+)
+from trinity.utils.xdg import (
+    is_under_xdg_trinity_root,
+)
+
+from .ropsten import (
+    RopstenLightChain,
+)
+
+
+def is_chain_initialized(chain_config):
+    """
+    - base dir exists
+    - chain data-dir exists
+    - nodekey exists and is non-empty
+    - canonical chain head in db
+    """
+    if not os.path.exists(chain_config.data_dir):
+        return False
+
+    if not os.path.exists(chain_config.database_dir):
+        return False
+
+    if chain_config.nodekey_path is None:
+        # has an explicitely defined nodekey
+        return True
```

Comment:
I don't understand what is happening here. Why is this short-circuiting if the path is none, and doesn't care if the chain is initialized?

Reply:
You're lack of understanding is because this is broken (and not covered in the tests apparently).  This logic is wrong.  `nodekey_path == None` indicates that the nodekey was explicitly specified.  This should be:

```python
if chain_config.nodekey_path is not None and not os.path.exists(chain_config.nodekey_path):
   return False
```

Will fix

Reply:
Good catch.  Should be `database_dir`.

Reply:
You're lack of understanding is because this is broken (and not covered in the tests apparently).  This logic is wrong.  `nodekey_path == None` indicates that the nodekey was explicitly specified.  This should be:

```python
if chain_config.nodekey_path is not None and not os.path.exists(chain_config.nodekey_path):
   return False
```

Will fix